json/zio/src/main/scala/sttp/tapir/json/zio/TapirJsonZio.scala

```diff
@@ -0,0 +1,43 @@
+package sttp.tapir.json.zio
+
+import sttp.tapir.Codec.JsonCodec
+import sttp.tapir.DecodeResult.{Error, Value}
+import sttp.tapir.SchemaType.{SCoproduct, SObjectInfo, SProduct}
+import sttp.tapir.{EndpointIO, Schema, anyFromUtf8StringBody}
+import zio.json.ast.Json
+import zio.json.ast.Json.Obj
+import zio.json.{JsonDecoder, JsonEncoder, _}
+
+import scala.collection.immutable.ListMap
+
+trait TapirJsonZio {
+
+  def jsonBody[T: JsonEncoder: JsonDecoder: Schema]: EndpointIO.Body[String, T] = anyFromUtf8StringBody(zioCodec[T])
+
+  implicit def zioCodec[T: JsonEncoder: JsonDecoder: Schema]: JsonCodec[T] =
+    sttp.tapir.Codec.json[T] { s =>
+      zio.json.JsonDecoder.apply[T].decodeJson(s) match {
+        case Left(error)  => Error(s, new IllegalArgumentException(error))
+        case Right(value) => Value(value)
+      }
+    } { t => t.toJson }
```

Comment:
Here's my two cents in parsing zio-json's error message into Tapir's json path:
```suggestion
  implicit def zioJsonCodec[T: JsonEncoder: JsonDecoder: Schema]: JsonCodec[T] =
    sttp.tapir.Codec.json[T] { s =>
      s.fromJson[T] match {
        case Left(msg) =>
          val (message, path) = parseErrorMessage(msg)
          Error(s, JsonDecodeException(List(JsonError(message, path)), new Exception(msg)))
        case Right(v) => Value(v)
      }
    } { t => t.toJson }

  private def parseErrorMessage(errorMessage: String): (String, List[FieldName]) = {
    val leftParenIndex = errorMessage.indexOf('(')
    if (leftParenIndex >= 0) {
      val path = errorMessage.substring(0, leftParenIndex)
      val message = errorMessage.substring(leftParenIndex + 1, errorMessage.length - 1)
      return message -> path.split("\\.").toList.filter(_.nonEmpty).map(FieldName.apply)
    } else {
      return errorMessage -> List.empty
    }
  }
```

Reply:
Thanks @alphaho. I will do the changes tonight ;)

Reply:
@alphaho let me know if you have any feedback ;)